foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiError.java

```diff
@@ -0,0 +1,31 @@
+package com.sap.ai.sdk.foundationmodels.openai;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.common.annotations.Beta;
+import com.sap.ai.sdk.core.common.ClientError;
+import com.sap.ai.sdk.foundationmodels.openai.model2.ErrorResponse;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import lombok.experimental.Delegate;
+
+/** Represents an error response from the OpenAI API. */
+@Beta
+@Value
+@AllArgsConstructor(onConstructor = @__({@JsonCreator}), access = AccessLevel.PROTECTED)
+public class OpenAiError implements ClientError {
```

Comment:
(Comment)

I'm unhappy about the interface name "ClientError", since "Error" is reserved for non-recoverable exceptions leading to application shutdown.